io/disk/disk_info.py

```diff
@@ -0,0 +1,242 @@
+            pkg = 'xfsprogs'
+        if self.fstype == 'btrfs':
+            if distro.detect().name == 'Ubuntu':
+                pkg = "btrfs-tools"
+        if pkg and not smm.check_installed(pkg) and not smm.install(pkg):
+            self.skip("Package %s is missing and could not be installed" % pkg)
+
+        pkg_list = ["lshw", "hwinfo"]
+        for pkg in pkg_list:
+            if pkg and not smm.check_installed(pkg) and not smm.install(pkg):
+                self.skip("Package %s is missing and could not be installed"
+                          % pkg)
+
+    def run_command(self, cmd):
+        """
+        Run command and fail the test if any command fails
+        """
+        try:
+            process.run(cmd, shell=True, sudo=True)
+        except CmdError as details:
+            self.fail("Command %s failed %s" % (cmd, details))
+
+    def test_commands(self):
+        """
+        Test block device tools to list different disk devices
+        """
+        cmd_list = ["lsblk -l", "fdisk -l", "sfdisk -l", "parted -l",
+                    "df -h", "blkid", "lshw -c disk", "hwinfo --block --short"]
+        for cmd in cmd_list:
+            self.run_command(cmd)
+
+    def test(self):
+        disk = (self.disk.split("/dev/"))[1]
+        if process.system("ls /dev/disk/by-id -l| grep -i %s" % disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in /dev/disk/by-id")
+        if process.system("ls /dev/disk/by-path -l| grep -i %s" % disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in /dev/disk/by-path")
+
+        # Verify disk listed in all tools
+        if process.system("fdisk -l %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in fdisk o/p")
+        if process.system("blkid %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in blkid o/p")
+        if process.system("lsblk %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in lsblk o/p")
+        if process.system("parted -l| grep -i %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in parted o/p")
+        if process.system("lshw -c disk | grep -i %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in lshw o/p")
+        if process.system("hwinfo --short --block | grep -i %s" % self.disk,
+                          ignore_status=True, shell=True, sudo=True) != 0:
+            self.fail("Given disk %s is not present in hwinfo o/p")
+
```

Comment:
better to have all the above commands in a list and call a single process.system(" <command> -l| grep -i %s" % self.disk, ignore_status=True, shell=True, sudo=True)  in a loop... looks better